### ISSUE TITLE
Support a standard socket interface

### ISSUE THREAD

**Issue by diondokter** [diondokter/nrf-modem#6]
Currently the APIs are all custom.

Once the [embedded-nal async](https://github.com/rust-embedded-community/embedded-nal/tree/master/embedded-nal-async) work is done, we should probably implement it.

**Comment by jue89** [comment]
Or maybe this one?

https://docs.rs/embedded-io-async/latest/embedded_io_async/

**Comment by diondokter** [comment]
Yeah, I'm not a big fan of the NAL api. So embedded-io is a good one to add too!

**Comment by diondokter** [comment]
Added in the 0.6.0 release!

**Comment by jue89** [comment]
Wow ... awesome! I like no time :D

**Comment by diondokter** [comment]
Alright, there doesn't seem to be a new non-nal option since 2022. Closing this. If there's a good option in the future, feel free to put it here!
